packages/discovery-react-components/src/components/StructuredQuery/utils/stringifyStructuredQuerySelection.ts

```diff
@@ -4,29 +4,31 @@ export const stringifyStructuredQuerySelection = (
   structuredQuerySelection: StructuredQuerySelection
 ) => {
   let stringifiedStructuredQuerySelection: string = '';
-  stringifiedStructuredQuerySelection += Object.keys(structuredQuerySelection.groups)
+  stringifiedStructuredQuerySelection += structuredQuerySelection.group_order
     .map(groupId => {
-      return groupId === '0'
+      return groupId === 0
         ? stringifyRows(structuredQuerySelection, groupId)
         : '(' + stringifyRows(structuredQuerySelection, groupId) + ')';
     })
     .join(structuredQuerySelection.groups[0].operator);
   return stringifiedStructuredQuerySelection;
 };
 
-const stringifyRows = (structuredQuerySelection: StructuredQuerySelection, groupId: string) => {
+const stringifyRows = (structuredQuerySelection: StructuredQuerySelection, groupId: number) => {
   return structuredQuerySelection.groups[groupId].rows
-    .map((rowId: string) => {
+    .map((rowId: number) => {
       const row = structuredQuerySelection.rows[rowId];
-      const rowField = containsReservedCharacters(row.field) ? '"' + row.field + '"' : row.field;
-      const rowValue = containsReservedCharacters(row.value) ? '"' + row.value + '"' : row.value;
+      const rowField = escapeReservedCharacters(row.field);
+      const rowValue = row.value === '' ? row.value : '"' + row.value + '"';
```

Comment:
Should we always be quoting values? I would think that we would only quote things that have reserved characters. Also we need to make sure we escape double quotes

Reply:
Yeah, that's what I was thinking too. But I updated to always quote values since I realized that's what we're doing on cloud and that way we don't spend time checking for reserved characters when the values can just always be quoted.

I have double quotes in the regex, or maybe I'm missing something?